packages/web-frontend/src/components/HierarchyItem.js

```diff
@@ -20,83 +20,73 @@
  * @return {element} a HierarchyItem react component.
  */
 
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import FlatButton from 'material-ui/FlatButton';
 import ClosedIcon from 'material-ui/svg-icons/navigation/chevron-right';
 import OpenIcon from 'material-ui/svg-icons/navigation/expand-more';
-import SelectedIcon from 'material-ui/svg-icons/toggle/radio-button-checked';
-import UnSelectedIcon from 'material-ui/svg-icons/toggle/radio-button-unchecked';
+import SelectedCheckBoxIcon from 'material-ui/svg-icons/toggle/check-box';
+import UnSelectedCheckBoxIcon from 'material-ui/svg-icons/toggle/check-box-outline-blank';
 import CircularProgress from '@material-ui/core/CircularProgress';
 import { ReferenceTooltip } from '@tupaia/ui-components';
 
-export class HierarchyItem extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isOpen: false,
-    };
-  }
-
-  onClick() {
-    if (this.props.onClick) {
-      this.props.onClick();
-    }
-    this.setState(state => ({ isOpen: !state.isOpen }));
-  }
-
-  renderOpenClosedIcon() {
-    const { nestedItems, hasNestedItems } = this.props;
-    const { isOpen } = this.state;
+export const HierarchyItem = React.memo(
+  ({
+    label,
+    style,
+    nestedMargin,
+    nestedItems,
+    isSelected,
+    Icon,
+    isLoading,
+    hasNestedItems,
+    info,
+    onClick,
+    dispatch,
+    ...otherProps
+  }) => {
+    const [isOpen, setIsOpen] = useState(false);
 
-    const hasChildren = hasNestedItems || (Array.isArray(nestedItems) && nestedItems.length > 0);
-    if (!hasChildren) {
-      return null;
-    }
+    const buttonOnClick = () => {
+      if (onClick) {
+        onClick();
+      }
+      setIsOpen(!isOpen);
+    };
 
-    const IconComponent = isOpen ? OpenIcon : ClosedIcon;
-    return <IconComponent style={styles.buttonIcon} />;
-  }
+    const OpenClosedIcon = () => {
```

Comment:
The convention we use across the codebase for elements rendered inside the same element is using `render...` methods:

```suggestion
    const renderOpenClosedIcon = () => {
```